base/LineEdit.jl

```diff
@@ -146,16 +148,14 @@ function complete_line(s::PromptState, repeats)
     elseif length(completions) == 1
         # Replace word by completion
         prev_pos = position(s.input_buffer)
-        seek(s.input_buffer, prev_pos-sizeof(partial))
```

Comment:
Probably because `edit_replace` already is "undo aware", whereas `seek` is not. When I compile without this change, after an undo the cursor is misplaced (e.g. at the beginning of a (prefix) word after undoing a completion).